video/out/hwdec/hwdec_vaapi.c

```diff
@@ -258,18 +264,40 @@ static int mapper_map(struct ra_hwdec_mapper *mapper)
     VAStatus status;
     VADisplay *display = p_owner->display;
     VADRMPRIMESurfaceDescriptor desc = {0};
-
+    bool wldmabuf = ra_is_wldmabuf(mapper->ra);
+    uint32_t flags = wldmabuf ? VA_EXPORT_SURFACE_COMPOSED_LAYERS : VA_EXPORT_SURFACE_SEPARATE_LAYERS;
     status = vaExportSurfaceHandle(display, va_surface_id(mapper->src),
                                    VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2,
                                    VA_EXPORT_SURFACE_READ_ONLY |
-                                   VA_EXPORT_SURFACE_SEPARATE_LAYERS,
+                                   flags,
                                    &desc);
     if (!CHECK_VA_STATUS_LEVEL(mapper, "vaExportSurfaceHandle()",
                                p_owner->probing_formats ? MSGL_DEBUG : MSGL_ERR))
     {
         close_file_descriptors(desc);
         goto err;
     }
+
+    if (mapper->owner->probing && wldmabuf) {
+        close_file_descriptors(desc);
+        /* composed has single layer */
+        int layer_no = 0;
+        uint32_t drm_format = desc.layers[layer_no].drm_format;
+
+        if (!ra_compatible_format(mapper->ra, drm_format, desc.objects[0].drm_format_modifier)) {
+            MP_DBG(mapper->owner, "Mapped surface with format %s; drm format '%s(%016lx)', is "
+                                  "not supported by compositor.\n",
+                                   mp_imgfmt_to_name(mapper->src->params.hw_subfmt),
+                                   mp_tag_str(drm_format),
+                                   desc.objects[0].drm_format_modifier);
+            goto err;
+        }
+        wayland_format wlformat = {drm_format, 0, desc.objects[0].drm_format_modifier};
+        MP_TARRAY_APPEND(p_owner, p_owner->supported_wl_formats, p_owner->num_supported_wl_formats, wlformat);
```

Comment:
what about the suggestion to move this to `try_format_pixfmt`?

Reply:
This code needs the mapper, so best I could do was move it into `try_format_map`.

Reply:
This has been moved into the dmabuf interop.